File: go/book-builder-strategy/depth/depth.go
package depth


type Depths []Depth

type Depth struct {
	Symbol        string  `json:"symbol"`
	MarketPercent float64 `json:"marketPercent"`
	Volume        int     `json:"volume"`
	LastSalePrice float64 `json:"lastSalePrice"`
	LastSaleSize  int     `json:"lastSaleSize"`
	LastSaleTime  int64   `json:"lastSaleTime"`
	LastUpdated   int64   `json:"lastUpdated"`
	Bids          []struct {
		Price     float64 `json:"price"`
		Size      int     `json:"size"`
		Timestamp int64   `json:"timestamp"`
	} `json:"bids"`
	Asks []struct {
		Price     float64 `json:"price"`
		Size      int     `json:"size"`
		Timestamp int64   `json:"timestamp"`
	} `json:"asks"`
	SystemEvent struct {
		SystemEvent string `json:"systemEvent"`
		Timestamp   int64  `json:"timestamp"`
	} `json:"systemEvent"`
	TradingStatus struct {
		Status    string `json:"status"`
		Reason    string `json:"reason"`
		Timestamp int64  `json:"timestamp"`
	} `json:"tradingStatus"`
	OpHaltStatus struct {
		IsHalted  bool  `json:"isHalted"`
		Timestamp int64 `json:"timestamp"`
	} `json:"opHaltStatus"`
	SsrStatus struct {
		IsSSR     bool   `json:"isSSR"`
		Detail    string `json:"detail"`
		Timestamp int64  `json:"timestamp"`
	} `json:"ssrStatus"`
	SecurityEvent struct {
		SecurityEvent string `json:"securityEvent"`
		Timestamp     int64  `json:"timestamp"`
	} `json:"securityEvent"`
	Trades []struct {
		Price                 float64 `json:"price"`
		Size                  int     `json:"size"`
		TradeID               int     `json:"tradeId"`
		IsISO                 bool    `json:"isISO"`
		IsOddLot              bool    `json:"isOddLot"`
		IsOutsideRegularHours bool    `json:"isOutsideRegularHours"`
		IsSinglePriceCross    bool    `json:"isSinglePriceCross"`
		IsTradeThroughExempt  bool    `json:"isTradeThroughExempt"`
		Timestamp             int64   `json:"timestamp"`
	} `json:"trades"`
	TradeBreaks []interface{} `json:"tradeBreaks"`
}

File: java/fixmarketsimulator/src/main/java/com/ettech/fixmarketsimulator/bookbuilder/Depth.java
package com.ettech.fixmarketsimulator.bookbuilder;

import java.util.List;



public class Depth{
    public String symbol;
    public double marketPercent;
    public int volume;
    public double lastSalePrice;
    public int lastSaleSize;
    public long lastSaleTime;
    public long lastUpdated;
    public List<Line> bids;
    public List<Line> asks;
    public SystemEvent systemEvent;
    public TradingStatus tradingStatus;
    public OpHaltStatus opHaltStatus;
    public SsrStatus ssrStatus;
    public SecurityEvent securityEvent;
    public List<Object> trades;
    public List<Object> tradeBreaks;

    public static class Line {
        public double price;
        public int size;
        public Object timestamp;
    }

    public static  class SystemEvent{
        public String systemEvent;
        public long timestamp;
    }

    public  static class TradingStatus{
        public String status;
        public String reason;
        public long timestamp;
    }

    public static  class OpHaltStatus{
        public boolean isHalted;
        public long timestamp;
    }

    public  static class SsrStatus{
        public boolean isSSR;
        public String detail;
        public long timestamp;
    }

    public  static  class SecurityEvent{
        public String securityEvent;
        public long timestamp;
    }
}
